src/Node/ReadLine.purs

```diff
@@ -14,29 +17,36 @@ foreign import data Interface :: *
 
 -- | A function which performs tab completion.
 -- |
--- | This function takes the partial command as input, and returns a collection of 
+-- | This function takes the partial command as input, and returns a collection of
 -- | completions, as well as the matched portion of the input string.
 type Completer eff = String -> Eff (console :: CONSOLE | eff) { completions :: Array String, matched :: String }
 
--- | A function which handles input from the user.
-type LineHandler eff a = String -> Eff (console :: CONSOLE | eff) a 
+-- | A completion function which offers no completions.
+noCompletion :: forall eff. Completer eff
+noCompletion s = return { completions: [], matched: s }
+
+-- | A function which handles each line of input.
+type LineHandler eff a = String -> Eff (console :: CONSOLE | eff) a
 
 -- | Set the current line handler function.
 foreign import setLineHandler :: forall eff a. Interface -> LineHandler eff a -> Eff (console :: CONSOLE | eff) Interface
 
--- | Prompt the user for input on the specified `Interface`.
-foreign import prompt :: forall eff. Interface -> Eff (console :: CONSOLE | eff) Interface
-
--- | Set the prompt.
-foreign import setPrompt :: forall eff. String -> Int -> Interface -> Eff (console :: CONSOLE | eff) Interface
-
--- | Create an interface with the specified completion function.
-foreign import createInterface :: forall eff. Completer eff -> Eff (console :: CONSOLE | eff) Interface
+-- | Options passed to `readline`'s `createInterface`
+data InterfaceOptions r w a e
+  = Input { input :: Readable r e }
```

Comment:
I think a single record type might work better here, using `Maybe` for the optional fields. Then, we could have another function in this module (the PureScript part, not the FFI part) which converts it to the actual representation expected by the node api. See [`purescript-node-child-process`](https://github.com/purescript-node/purescript-node-child-process/blob/master/src/Node/ChildProcess.purs#L184-L198), where I did a similar thing. That would be a good way of avoiding the `value0` stuff, I think.